sdk/storage/storage-blob/src/utils/utils.common.ts

```diff
@@ -224,7 +224,10 @@ export function extractConnectionStringParts(connectionString: string): Connecti
     // SAS connection string
 
     let accountSas = getValueInConnString(connectionString, "SharedAccessSignature");
-    const accountName = getAccountNameFromUrl(blobEndpoint);
+    let accountName = getValueInConnString(connectionString, "AccountName");
```

Comment:
Main change is here: Try to retrieve the AccountName from the connection string first instead of trying to deduce it from the BlobEndpoint